### DISCUSSION TITLE
Integration of a 3D dataset along one axis to get a 2D dataset (3D projection along one axis) using Pyvista

### DISCUSSION THREAD

**Discussion by BobHeidsieck** [pyvista/pyvista#3860] [Q&A] [answered]
Hello,
I am using finite element method to manipulate anatomical structure. I have found Pyvista quite convenient to manage this as an unstructured grid. I would like to simulate an x-ray image of these structures. This is could be accomplished by integrating all slices along one axis and perform the 2D np array transformation (see https://github.com/pyvista/pyvista-support/issues/89).
In other word is there an method in Pyvista that enable to integrate a 3D dataset along one axis to get a 2D dataset?
Using several slices and performing the 2D np transform would be a sub optimal option due to multiple sampling.
Thanks for any advice.

**Comment by banesullivan** [comment, marked as answer]
Generally speaking, no we do not have anything for doing this with general 3D datasets. Though we could implement something to do this for `pyvista.UniformGrid` (`vtkImageData`).

I imagine VTK actually has this functionality somewhere, it's just a matter of finding it.

**Reply by MatthewFlamm** [reply]
This seems like it could be a rabbit hole of possibilities to support in PyVista if it doesn't exist in VTK. For example, average value on slices along an axis.  Total flux on slices, etc.

If it doesn't exist in VTK, I think one or two cookbook examples in the documentation would get users moving in the right direction quickly.

**Comment by ARostekMU** [comment]
Is there any progress in this area? I would like to do a very similar thing.
